synapseclient/core/upload/multipart_upload.py

```diff
@@ -315,6 +323,7 @@ def _upload_parts(self, part_count, remaining_part_numbers):
                     executor.submit(
                         self._handle_part,
                         part_number,
+                        context.get_current(),
```

Comment:
Ah, I see. a bit inconvenient with the way threading is in Python, but this makes sense